app/models/manageiq/providers/azure/cloud_manager.rb

```diff
@@ -73,4 +77,80 @@ def vm_restart(vm, _options = {})
   rescue => err
     _log.error "vm=[#{vm.name}], error: #{err}"
   end
+
+  # Discovery
+
+  # Create EmsAzure instances for all regions with instances
+  # or images for the given authentication.  Created EmsAzure instances
+  # will automatically have EmsRefreshes queued up.  If this is a greenfield
+  # discovery, we will at least add an EmsAzure for eastus
+  def self.discover(clientid, clientkey, azure_tenant_id)
+    new_emses = []
+
+    all_emses = includes(:authentications)
+    all_ems_names = all_emses.index_by(&:name)
+
+    known_emses = all_emses.select { |e| e.authentication_userid == clientid }
+    known_ems_regions = known_emses.index_by(&:provider_region)
+
+    config     = raw_connect(clientid, clientkey, azure_tenant_id)
+    azure_vmm  = ::Azure::Armrest::VirtualMachineService.new(config)
+
+    azure_vmm.locations.each do |region|
+      region = region.delete(' ').downcase
+      next if known_ems_regions.include?(region)
+      next if vms_in_region(azure_vmm, region).count == 0 # instances
+      # TODO: Check if images are == 0 and if so then skip
```

Comment:
@blomquisg - correct. The azure-armrest gem 'get_images' method is not in place yet.